### PR TITLE
[SC-140] multiple synapse bucket owners

### PR DESCRIPTION
**Pull request by zaro0508**
With the fix to PLFM-6358 we can now refactor to allow users to
provision Synapse bucket with multiple owners.

### CODE REVIEW

File: s3/sc-s3-synapse-ra.yaml [hunk 1]
```diff
@@ -13,13 +13,17 @@ Metadata:
...
+          - SynapseIDs
           - S3UserARNs
 Parameters:
-  SynapseUserName:
-    Type: String
-    Description: Synapse user for bucket
+  SynapseIDs:
+    Type: CommaDelimitedList
+    Description: >-
+      (Optional) List of Synapse bucket user or team owners
```
Comment (tthyer) [inline s3/sc-s3-synapse-ra.yaml:22]: This can't be optional, as it is required to link to synapse.

Comment (brucehoff) [reply s3/sc-s3-synapse-ra.yaml:22]: Yes, I agree with Tess.

Comment (zaro0508) [reply s3/sc-s3-synapse-ra.yaml:22]: I disagree.  Users should be able to provision a bucket without a synapse ID.  They can just update this field with one or more synapse IDs at a later time.

Comment (tthyer) [reply s3/sc-s3-synapse-ra.yaml:22]: But why would anyone choose to provision a Synapse bucket without linking it when there's also the choice for a non-Synapse bucket? The users don't know how this works behind the scenes. If you let them provision a Synapse bucket without specifying the owner, the linkup with Synapse won't work and they won't know why.

Comment (zaro0508) [reply s3/sc-s3-synapse-ra.yaml:22]: RTFD? 😄

**Pull request merged**